ccl/multiregionccl: TestEnsureLocalReadsOnGlobalTables failed [repros]

ccl/multiregionccl.TestEnsureLocalReadsOnGlobalTables [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9941605?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9941605?buildTab=artifacts#/) on master @ [918923738a73c39986c163465edb70e9af2c49b8](https://github.com/cockroachdb/cockroach/commits/918923738a73c39986c163465edb70e9af2c49b8):


```
=== RUN   TestEnsureLocalReadsOnGlobalTables
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestEnsureLocalReadsOnGlobalTables3419223839
    test_log_scope.go:79: use -show-logs to present logs inline
    roundtrips_test.go:145: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/roundtrips_test.go:145
        	Error:      	Not equal: 
        	            	expected: true
        	            	actual  : false
        	Test:       	TestEnsureLocalReadsOnGlobalTables
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestEnsureLocalReadsOnGlobalTables3419223839
--- FAIL: TestEnsureLocalReadsOnGlobalTables (25.66s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEnsureLocalReadsOnGlobalTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-27657

--- a/pkg/ccl/multiregionccl/roundtrips_test.go
+++ b/pkg/ccl/multiregionccl/roundtrips_test.go
@@ -21,6 +21,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
@@ -34,6 +35,7 @@ import (
 func TestEnsureLocalReadsOnGlobalTables(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	skip.UnderStressRace(t, "https://github.com/cockroachdb/cockroach/issues/102798#issuecomment-1543852311")
 
 	// ensureOnlyLocalReads looks at a trace to ensure that reads were served
 	// locally. It returns true if the read was served as a follower read.
@@ -142,7 +144,7 @@ func TestEnsureLocalReadsOnGlobalTables(t *testing.T) {
 
 		// Expect every non-leaseholder to serve a (local) follower read. The
 		// leaseholder on the other hand won't serve a follower read.
-		require.Equal(t, !isLeaseHolder, followerRead)
+		require.Equal(t, !isLeaseHolder, followerRead, "%v", rec)
 	}
 
 	close(stopWritesCh)

